sources/src/parser/gmlfeaturecollectionparser.cpp

```diff
@@ -19,6 +20,12 @@ namespace citygml {
         : GMLObjectElementParser(documentParser, factory, logger)
     {
         m_bounds = nullptr;
+        m_srsSRSOverride = false;
+        std::string paramsSrcSRS = documentParser.getParserParams().srcSRS;
+        if (!paramsSrcSRS.empty()) {
+            m_bounds = new Envelope(paramsSrcSRS);
```

Comment:
doesn't this cause the complete envelope to be discarded (included the coordinates included in the envelope) ? Wouldn't it be more precise to apply only the srs override to existing envelopes instead of replacing them alls ?

Reply:
From what I see, the SRS is only overridden in the constructor before any envelope is created for this object element. So at this point there is no envelope with bounds right?

I'm happy to edit this to be more suitable. I'm still no expert in this codebase, so this was the best and cleanest way I thought this feature could work.

Also, I think the idea was to override all the SRS in the file with a single one. Basically, for files that have an SRS but don't specify it in the file itself, like I mentioned in the PR description.